sn_networking/src/record_store.rs

```diff
@@ -70,14 +70,14 @@ const MIN_STORE_COST: u64 = 1;
 
 /// A `RecordStore` that stores records on disk.
 pub struct NodeRecordStore {
-    /// The identity of the peer owning the store.
-    local_key: KBucketKey<PeerId>,
     /// The address of the peer owning the store
     local_address: NetworkAddress,
     /// The configuration of the store.
     config: NodeRecordStoreConfig,
-    /// A set of keys, each corresponding to a data `Record` stored on disk.
+    /// Main records store remains unchanged for compatibility
     records: HashMap<Key, (NetworkAddress, RecordType)>,
+    /// Additional index organizing records by distance bucket
+    records_by_bucket: HashMap<u32, HashSet<Key>>,
```

Comment:
I think we don't need separate fields of `records_by_bucket` and `records`, they can be merged into:
`records: HashMap<u32, HashMap<Key, (NetworkAddress, RecordType)>>`
during `key` query, the extra effor of calculating kbucket index will be compenstated by break a large map into smaller ones

and above all, it avoids hassale of `have to sync two maps` (which could be vulerable to bugs)